MetaU Capstone Project/API/SpotifyManager.m

```diff
@@ -105,9 +113,78 @@ - (void)appRemote:(nonnull SPTAppRemote *)appRemote didFailConnectionAttemptWith
 - (void)playerStateDidChange:(nonnull id<SPTAppRemotePlayerState>)playerState {
     NSLog(@"Track name: %@", playerState.track.name);
     NSLog(@"player state changed");
+    
+//    [MusicSessionViewController updateMusicSession];
+//    [MusicSessionViewController ];
+}
+
+
+- (id<SPTAppRemoteTrack>) getCurrentTrackInfo {
+    return self.currentTrack;
+}
+
+- (void)startTrack {
+//    NSLog(@"Play music called");
+    [[self.appRemote playerAPI] resume:^(id result, NSError * error){
+        NSLog(@"Playing current track...");
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
+        } else {
+            NSLog(@"Played music");
+        }
+    }];
+}
+
+- (void)stopTrack {
+    NSLog(@"%@", @([self.appRemote isConnected]));
+    [[self.appRemote playerAPI] pause:^(id result, NSError * error){
+        NSLog(@"Pausing current track...");
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
+        } else {
+            NSLog(@"Stopped music.");
+        }
+    }];
+}
+
+- (void)skipTrack {
+    [[self.appRemote playerAPI] skipToNext:^(id result, NSError * error){
+        NSLog(@"Skipping current track...");
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
+        } else {
+            NSLog(@"Skipped to next track");
+        }
+    }];
+    
+    [self.appRemote.playerAPI skipToNext:(nil)];
+}
+
+- (void)rewindTrack {
+    NSLog(@"Rewinding music called");
+    
+    self.timestamp = 8; // Placeholder timestamp
+    
+    if (self.timestamp < MAX_MILISECONDS) { // if current timestamp < x seconds, restart current song
```

Comment:
you would need to create a MIN_MILLISECONDS variable right? for a lower threshold?